catalog/core/catalog-core-api/src/main/java/ddf/catalog/source/SourceSupportedAttributes.java

```diff
@@ -0,0 +1,65 @@
+package ddf.catalog.source;
+
+import ddf.catalog.data.MetacardType;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Optionally define supported attributes for querying a particular {@link Source} instance and make
+ * the information available outside the system to support validation scenarios prior to submitting
+ * a query.
+ *
+ * <p>A set of queryable attributes is implied by this interface, which is defined as the collective
+ * set of all attributes across all supported metacard types, defined by {@link
+ * #getSupportedMetacardTypes()}, with the exception of any attributes listed as unsupported,
+ * defined by {@link #getUnsupportedAttributes()}.
+ *
+ * <p>Configuring supported attributes can be done using JSON files within {@code
+ * DDF_HOME/etc/sources/supported.json}, for example:
```

Comment:
I thought the supported attributes would be defined in code - near the code that actually handles translating from MAP to the source field schema